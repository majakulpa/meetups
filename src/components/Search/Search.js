import React from 'react'

const Search = ({ value, searchHandleChange }) => {
  return (
    <React.Fragment>
      <input
        className="shadow appearance-none border border-gray-500 rounded  py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
        value={value}
        onChange={searchHandleChange}
        placeholder="Search by event name or location"
      />
    </React.Fragment>
  )
}

export default Search
