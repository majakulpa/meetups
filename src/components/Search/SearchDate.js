import React from 'react'

const SearchDate = ({ date, searchDateHandleChange }) => {
  return (
    <React.Fragment>
      <input
        className="shadow appearance-none border border-gray-500 rounded py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
        value={date}
        onChange={searchDateHandleChange}
        type="date"
        min={new Date()
          .toISOString()
          .split('')
          .slice(0, 10)
          .join('')}
        placeholder="Enter date"
      />
    </React.Fragment>
  )
}

export default SearchDate
