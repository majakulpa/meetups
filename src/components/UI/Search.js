import React from 'react'
import { HiX, HiOutlineSearch } from 'react-icons/hi'

const Search = ({
  value,
  searchHandleChange,
  placeholder,
  handleClearSearch
}) => {
  return (
    <React.Fragment>
      <div className="relative">
        <input
          className="transition-colors duration-100 ease-in-out text-gray-600 py-2 
        pr-10 pl-8 block w-full appearance-none leading-normal border border-transparent 
        rounded focus:outline-none text-left select-none truncate focus:bg-white 
        focus:border-gray-300 bg-gray-200"
          value={value}
          onChange={searchHandleChange}
          placeholder={placeholder}
          id="search"
        />
        <div className="absolute inset-y-0 left-0 pl-2 flex items-center">
          <HiOutlineSearch />
        </div>
        <div
          className="cursor-pointer absolute inset-y-0 right-0 pr-4 flex items-center"
          onClick={handleClearSearch}
        >
          <HiX />
        </div>
      </div>
    </React.Fragment>
  )
}

export default Search
