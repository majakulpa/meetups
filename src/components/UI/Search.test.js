import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup, fireEvent } from '@testing-library/react'
import Search from './Search'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Search />, div)
})

test('if matches snapshot', () => {
  const tree = renderer.create(<Search />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('if clicking the button calls event handler once', () => {
  const mockHandler = jest.fn()

  const component = render(
    <Search
      placeholder="Search"
      value="Test"
      searchHandleChange={mockHandler}
      handleClearSearch={mockHandler}
    />
  )

  const div = component.getByPlaceholderText('Search')
  fireEvent.click(div)

  expect(div).toHaveTextContent('')
})
