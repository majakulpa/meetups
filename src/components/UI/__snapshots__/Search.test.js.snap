// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<div
  className="relative"
>
  <input
    className="transition-colors duration-100 ease-in-out text-gray-600 py-2  pr-10 pl-8 block w-full appearance-none leading-normal border border-transparent  rounded focus:outline-none text-left select-none truncate focus:bg-white  focus:border-gray-300 bg-gray-200"
    id="search"
    onChange={undefined}
    placeholder={undefined}
    value={undefined}
  />
  <div
    className="absolute inset-y-0 left-0 pl-2 flex items-center"
  >
    <svg
      className={undefined}
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="cursor-pointer absolute inset-y-0 right-0 pr-4 flex items-center"
    onClick={undefined}
  >
    <svg
      className={undefined}
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</div>
`;
